src/components/spinner/Spinner.tsx

```diff
@@ -0,0 +1,126 @@
+/*
+ * Echoes React
+ * Copyright (C) 2023-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+import { keyframes } from '@emotion/react';
+import styled from '@emotion/styled';
+import { useIntl } from 'react-intl';
+import { srOnly } from '~common/helpers/styles';
+
+interface Props {
+  ariaLabel?: string;
+  className?: string;
+  children?: React.ReactNode;
+  customSpinner?: JSX.Element;
+  isLoading?: boolean;
+  hasPlaceholder?: boolean;
+}
+
+export function Spinner(props: Readonly<Props>) {
+  const intl = useIntl();
+  const {
+    customSpinner,
+    className,
+    children,
+    hasPlaceholder,
+    ariaLabel = intl.formatMessage({
+      id: 'loading',
+      defaultMessage: 'Loading...',
+      description: 'aria-label text, to indicate that there is spinner rotating in this place',
+    }),
+    isLoading = true,
+  } = props;
+
+  if (customSpinner) {
+    return <>{isLoading ? customSpinner : children}</>;
+  }
```

Comment:
Do we have a legitimate use case for a custom spinner? 
In SQ we use the prop once, and it's to wrap the spinner in a `tr` + `td`, which we could probably do differently (maybe a `wrapper` prop?).
In that specific case, we're also losing the a11y features :unamused:

Reply:
Yeah I think that's a good point, I just checked in SC and it's the same we need a wrapper instead. The other cases can be easily rewritten.